### PR TITLE
import celery lazilly in pytest plugin and unignore flake8 F821,  "undefined name '...'"

### LINKED ISSUE #6283: unignore    `F821,  # undefined name '...'`
F821 is one of the most useful flake8 rules, however it's currently disabled because type comments reference undefined names.

it can be fixed with a conditional dependency on `typing` or importing typing in an `if False:` block, see discussion here: https://github.com/PyCQA/pyflakes/issues/501

# conditional dep
```
# requirements.txt
typing; python_version < '3.5'
```

```python
# some_annotated.py
from typing import Any, Tuple

if typing.TYPE_CHECKING:
    from celery import Celery
```

# `if False:`
```python
# some_annotated.py

if False:
    from typing import Any, Tuple
    from celery import Celery
```

@auvipy @thedrow do you have a preferred `typing` import approach? mine is to add the conditional dep

# Checklist

- [ ] I have checked the [issues list](https://github.com/celery/celery/issues?q=is%3Aissue+label%3A%22Issue+Type%3A+Enhancement%22+-label%3A%22Category%3A+Documentation%22)
  for similar or identical enhancement to an existing feature.
- [ ] I have checked the [pull requests list](https://github.com/celery/celery/pulls?q=is%3Apr+label%3A%22Issue+Type%3A+Enhancement%22+-label%3A%22Category%3A+Documentation%22)
  for existing proposed enhancements.
- [ ] I have checked the [commit log](https://github.com/celery/celery/commits/master)
  to find out if the if the same enhancement was already implemented in the
  master branch.
- [ ] I have included all related issues and possible duplicate issues in this issue
      (If there are none, check this box anyway).

#### Related Issues

- None

#### Possible Duplicates

- None

# Brief Summary

# Design

## Architectural Considerations

None

## Diagrams

N/A

## Alternatives

None

### PR DESCRIPTION
**Pull request by graingert**
fixes #6283

### CODE REVIEW

File: celery/platforms.py [hunk 1]
```diff
@@ -619,21 +627,8 @@ class Signals:
     ignored = _signal.SIG_IGN
     default = _signal.SIG_DFL
 
-    if hasattr(_signal, 'setitimer'):
-
-        def arm_alarm(self, seconds):
-            _signal.setitimer(_signal.ITIMER_REAL, seconds)
-    else:  # pragma: no cover
-        try:
-            from itimer import alarm as _itimer_alarm
```
Comment (graingert) [inline celery/platforms.py:628]: seems itimer is dead:

```
  Downloading itimer-1.0.tar.gz (26 kB)
    ERROR: Command errored out with exit status 1:
     command: /home/graingert/.virtualenvs/example/bin/python3.8 -c 'import io, os, sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-qynw_h4g/itimer_b50715d06a64406190606a9f7e376088/setup.py'"'"'; __file__='"'"'/tmp/pip-install-qynw_h4g/itimer_b50715d06a64406190606a9f7e376088/setup.py'"'"';f = getattr(tokenize, '"'"'open'"'"', open)(__file__) if os.path.exists(__file__) else io.StringIO('"'"'from setuptools import setup; setup()'"'"');code = f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-pm02m1yo
         cwd: /tmp/pip-install-qynw_h4g/itimer_b50715d06a64406190606a9f7e376088/
    Complete output (19 lines):
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-qynw_h4g/itimer_b50715d06a64406190606a9f7e376088/setup.py", line 6, in <module>
        import paver.tasks
      File "<frozen importlib._bootstrap>", line 991, in _find_and_load
      File "<frozen importlib._bootstrap>", line 971, in _find_and_load_unlocked
      File "<frozen importlib._bootstrap>", line 914, in _find_spec
      File "<frozen importlib._bootstrap_external>", line 1407, in find_spec
      File "<frozen importlib._bootstrap_external>", line 1381, in _get_spec
      File "<frozen importlib._bootstrap_external>", line 1362, in _legacy_get_spec
      File "<frozen importlib._bootstrap>", line 414, in spec_from_loader
      File "<frozen importlib._bootstrap_external>", line 709, in spec_from_file_location
      File "<frozen zipimport>", line 191, in get_filename
      File "<frozen zipimport>", line 713, in _get_module_code
      File "<frozen zipimport>", line 647, in _compile_source
      File "paver-minilib.zip/paver/tasks.py", line 68
        print output
              ^
    SyntaxError: Missing parentheses in call to 'print'. Did you mean print(output)?
    ----------------------------------------

```

File: celery/contrib/pytest.py [hunk 2]
```diff
@@ -1,9 +1,13 @@
 """Fixtures and testing utilities for :pypi:`pytest <pytest>`."""
 import os
 from contextlib import contextmanager
+from typing import Any, Mapping, Sequence, Union
 
 import pytest
 
+from celery import Celery
```
Comment (graingert) [inline celery/contrib/pytest.py:8]: urgh the pytest plugin is importing celery before coverage checking is enabled

File: celery/contrib/pytest.py [hunk 3]
```diff
@@ -1,11 +1,17 @@
 """Fixtures and testing utilities for :pypi:`pytest <pytest>`."""
 import os
 from contextlib import contextmanager
+from typing import TYPE_CHECKING, Any, Mapping, Sequence, Union
 
 import pytest
 
-from .testing import worker
-from .testing.app import TestApp, setup_default_app
```
Comment (graingert) [inline celery/contrib/pytest.py:8]: importing these lazilly increased coverage by `(+9.96%)`!

**Comment by graingert** [comment]
@auvipy time to up the coverage target !

**Review by thedrow — APPROVED** [review]
I knew something was wrong with our coverage measurement, but I wasn't sure what made the coverage percentage drop so much.

Thanks!

**Pull request merged**